lib/mayaUsd/ufe/UsdContextOps.cpp

```diff
@@ -54,6 +56,58 @@ class SetVariantSelectionUndoableCommand : public Ufe::UndoableCommand
     const std::string fNewSelection;
 };
 
+//! \brief Undoable command for add new prim
+class AddNewPrimUndoableCommand : public Ufe::UndoableCommand
+{
+public:
+    AddNewPrimUndoableCommand(const MayaUsd::ufe::UsdSceneItem::Ptr& usdSceneItem,
+                              const Ufe::ContextOps::ItemPath& itemPath)
+    : Ufe::UndoableCommand()
+    {
+        // First get the stage from the proxy shape.
+        auto ufePath = usdSceneItem->path();
+        auto segments = ufePath.getSegments();
+        auto dagSegment = segments[0];
+        _stage = MayaUsd::ufe::getStage(Ufe::Path(dagSegment));
+        if (_stage) {
+            // Rename the new prim for uniqueness, if needed.
+            Ufe::Path newUfePath = ufePath + itemPath[1];
```

Comment:
You've tried this, so clearly it works, but I'm just not sure how :)  In the parenting code, when I parent just below the proxy shape, I needed to take care of the fact that the newly-added component is in a different run-time:
https://github.com/Autodesk/maya-usd/blob/dev/lib/mayaUsd/ufe/UsdUndoInsertChildCommand.cpp#L53
Why don't you have the same problem?  If you just concatenate the prim name to the existing proxy shape UFE path, you'll get a path that is in the Maya run-time, which is incorrect.

Reply:
I think it's O.K. like this, but I'd add in a comment somehow to say something like "Hey I know this isn't 100% correct, but all uniqueName does is take the back() of the path, so it's fine."

Reply:
Done. I also added another code comment for the visibility.